File: plugin/src/main/java/com/atsoft/jira/plugin/logviewer/impl/MyPluginComponentImpl.java
package com.atsoft.jira.plugin.logviewer.impl;

import com.atlassian.plugin.spring.scanner.annotation.export.ExportAsService;
import com.atlassian.plugin.spring.scanner.annotation.imports.ComponentImport;
import com.atlassian.sal.api.ApplicationProperties;
import com.atsoft.jira.plugin.logviewer.api.MyPluginComponent;

import jakarta.inject.Inject;
import jakarta.inject.Named;

@ExportAsService({ MyPluginComponent.class })
@Named("myPluginComponent")
public class MyPluginComponentImpl implements MyPluginComponent {
    @ComponentImport
    private final ApplicationProperties applicationProperties;

    @Inject

    public MyPluginComponentImpl(final ApplicationProperties applicationProperties) {
        this.applicationProperties = applicationProperties;
    }

    public String getName() {
        if (null != applicationProperties) {
            return "myComponent:" + applicationProperties.getDisplayName();
        }

        return "myComponent";
    }
}

File: plugin/src/main/kotlin/com/atsoft/jira/plugin/logviewer/impl/MyPluginComponentImpl.kt
package com.atsoft.jira.plugin.logviewer.impl;

import com.atsoft.jira.plugin.logviewer.api.MyPluginComponent;
import com.atlassian.plugin.spring.scanner.annotation.export.ExportAsService;

@ExportAsService(MyPluginComponent::class)
class MyPluginComponentImpl : MyPluginComponent {
    override val name: String? = "myComponent"
}